tests/pass/float.rs

```diff
@@ -148,155 +213,268 @@ fn basic() {
+        assert_ftoi(5.9, 5, "5.0");
+        assert_ftoi_neg(-5.9, izero.saturating_sub(5), "-5.0");
+
+        // Exercise the middle of the integer's bit range. A power of two fits as long as the
+        // exponent can fit its log2, so cap at the maximum representable power of two (which
+        // is the exponent's bias).
+        let half_i_max: $ity = 1 << min(ibits / 2, <$fty>::EXPONENT_BIAS);
+        let half_i_min = izero.saturating_sub(half_i_max);
+        assert_bidir(half_i_max as $fty, half_i_max, "half int max");
+        assert_bidir(half_i_min as $fty, half_i_min, "half int min");
+
+        // Integer limits
+        assert_bidir_big(imax_f, imax, "i max");
+        assert_bidir_big(imin_f, imin, "i min");
+        assert_ftoi_big(imax_f + 0.99, <$ity>::MAX, "slightly above i max");
+        assert_ftoi_neg(imin_f - 0.99, <$ity>::MIN, "slightly below i min");
+
+        if fbits == ibits {
+            // Signed integers have one fewer bit of range
+            let adj = if isigned { 1 } else { 0 };
+
+            // When F and I have the same number of bits... I have no clue
+            let nearest_rep_offset_u32 = 1 << (<$fty>::EXPONENT_BITS - 1 - adj);
+            let imax_offset_u32 = <$fty>::EXPONENT_MAX >> adj;
+
+            // Convert values to integers
+            let nearest_rep_offset = <$ity>::try_from(nearest_rep_offset_u32).unwrap();
+            let imax_offset = <$ity>::try_from(imax_offset_u32).unwrap();
+
+            // Max representable value that is <= I::MAX
+            let max_le_imax = imax - nearest_rep_offset;
+            assert_bidir(max_le_imax as $fty, imax - imax_offset, "max representable near int max");
+
+            // The next value up is unrepresentable
+            assert_bidir_unrep((max_le_imax + 1) as $fty, imax, "one more than max representable");
+
+            if isigned {
+                // Min representable value that is <= I::MIN
+                let min_ge_imin = imin + nearest_rep_offset;
+                assert_bidir(min_ge_imin as $fty, imin, "min representable near int min");
+                assert_bidir((min_ge_imin - 1) as $fty, imin, "min representable near int min");
+            }
+        }
+
+        if fbits >= ibits {
+            // Max representable power of 10
+            let pow10_max = (10 as $ity).pow(imax.ilog10());
+            assert_bidir(pow10_max as $fty, pow10_max, "pow10 max");
+
+            // Float limits
+            assert_ftoi_unrep(<$fty>::MAX, imax, "f max");
+            assert_ftoi_unrep(<$fty>::MIN, imin, "f min");
+        }
+
+        if ibits > fbits * 2 && !isigned {
+            // Unsigned max is enough to saturate only when more than 2x bit width
+            assert_itof(imax, <$fty>::INFINITY, "imax, ibits > fbits * 2");
+        } else {
+            // Signed values and anything smaller still fit
+            assert_itof(imax, imax_f, "imax, ibits > fbits * 2");
```

Comment:
Why does this say `ibits > fbits * 2`? It's in the `else` branch!

Also why can things never saturate for `signed`?

Reply:
I guess this `if` branch isn't even needed, if `imax as F` saturates then the literal is also saturated. Removed